src/widget/wwidgetstack.cpp

```diff
@@ -85,13 +96,32 @@ void WWidgetStack::hideIndex(int index) {
     }
 }
 
+void WWidgetStack::showEvent(QShowEvent*) {
+    int index = static_cast<int>(m_currentPageControl.get());
+    setCurrentIndex(index);
+
+    // Set the page triggers to match the current index.
+    for (QMap<int, ControlObject*>::iterator it = m_triggers.begin();
+            it != m_triggers.end(); ++it) {
+        it.value()->set(it.key() == index ? 1.0 : 0.0);
+    }
+}
+
 void WWidgetStack::onNextControlChanged(double v) {
+    // Don't do anything if we are not visible.
+    if (!isVisible()) {
+        return;
+    }
     if (v > 0.0) {
         setCurrentIndex((currentIndex() + 1) % count());
```

Comment:
This assumes that we have consecutive number of pages. The code above in WWidgetStack::showEvent allows gaps. 
Both versions works for me, but it should be consistent.